### PR TITLE
Add deprecation notice to README

### PR DESCRIPTION
**Pull request by rossf7**
Towards https://github.com/giantswarm/giantswarm/issues/7584

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 # e2e-harness
 
+**Deprecated** We are moving away from using e2e-harness in our e2e tests.
+We now provision KinD clusters instead of using minikube. This is done by [e2ectl](https://github.com/giantswarm/e2ectl).
```
Comment (rossf7) [inline README.md]: @sslavic I know you're still working on KinD to use a Circle Orb. We can update this if things change.

Comment (sslavic) [reply README.md]: Yep, will remember to update it once reusable e2e-integration-test job is in architect-orb.

**Review by sslavic — APPROVED** [review]
LGTM

**Review by MarcelMue — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (rossf7) [reply README.md]: @sslavic Great, thx!

Comment (kopiczko) [reply README.md]: If we plan to use the orb long term what is the reason to not use kind binary directly?

Comment (rossf7) [reply README.md]: As long as we can still provide a config file to kind I'm fine with using the binary directly.

We use this currently for multi node clusters and we may need more settings in future. e.g extending the kubeadm config to enable feature gates. 

https://github.com/giantswarm/e2ectl/blob/451a0b4a6f6d020683bcc2999d79d85aa5e2ec1e/cmd/cluster/create/template.go

Comment (kopiczko) [reply README.md]: That doesn't seem to be a problem to me. Thanks for the heads up.

**Pull request merged**